src/core/lifetime.d

```diff
@@ -2361,3 +2368,66 @@ template _d_delstructImpl(T)
     assert(innerDtors == 2);
     assert(outerDtors == 1);
 }
+
+/**
+ * Allocate an exception of type `T` from the exception pool.
+ * It has the same interface as `rt.lifetime._d_newclass()`.
+ * The class type must be Throwable or derived from it,
+ * and cannot be a COM or C++ class. The compiler must enforce this.
+ * Returns:
+ *      default initialized instance of the type
+ */
+Throwable _d_newThrowable(T)() @trusted
+    if (is(T : Throwable))
+{
+    auto ci = T.classinfo;
+
+    debug(PRINTF) printf("_d_newThrowable(ci = %p, %s)\n", ci, cast(char *) T.stringof);
+
+    assert(!(ci.m_flags & TypeInfo_Class.ClassFlags.isCOMclass));
+    assert(!(ci.m_flags & TypeInfo_Class.ClassFlags.isCPPclass));
+
+    import core.stdc.stdlib : malloc;
+    auto init = ci.initializer;
```

Comment:
I think you can use `T.init` here.

Reply:
Should probably use `__traits(initSymbol, T)` (and fall back to TypeInfo during CTFE).

Reply:
I don't have to use `TypeInfo` for CTFE either, because the lowering looks as specified here: https://github.com/dlang/dmd/pull/13494#issue-1094640159, so the hook doesn't reach CTFE.